AwsCryptographicMaterialProviders/dafny/AwsCryptographyKeyStoreAdmin/src/Mutations.dfy

```diff
@@ -87,10 +93,38 @@ module {:options "/functionSyntax:4" } Mutations {
     modifies keyManagerStrategy.Modifies
     ensures keyManagerStrategy.ValidState()
   {
-    var kmsOperation: string;
+
     var success?: bool := false;
     var throwAwayError;
-
+    // TODO-HV-2-M3: Support mutations on HV-2 item (mutation starting with hv-2 item)
+    :- Need(
+      item.EncryptionContext[Structure.HIERARCHY_VERSION] == Structure.HIERARCHY_VERSION_VALUE_1,
+      Types.KeyStoreAdminException(
+        message := "At this time, Mutations ONLY support HV-1; BK's Active Item is HV-2.")
+    );
+    if (isTerminalHv2?) {
+      // TODO-HV-2-M2: Add test to cover the if condition of this code path
+      // TODO-HV-2-M4: Support other key manager strategy
+      :- Need(keyManagerStrategy.kmsSimple?, Types.KeyStoreAdminException(message:="only KMS Simple allow when mutating to hv-2."));
```

Comment:
```suggestion
      :- Need(keyManagerStrategy.kmsSimple?, Types.KeyStoreAdminException(message:="Only keyManagerStrategy.kmsSimple is allowed when mutating to hv-2."));
```

Reply:
or replace it with the name we have in the smithy model

Reply:
Done. Wrote the name from smithy model. 

[020c0b3](https://github.com/aws/aws-cryptographic-material-providers-library/pull/1442/commits/020c0b308c2cb7382c8db461d415db40be7fe63a)

Reply:
And had to update the test as well. [4e5d7af](https://github.com/aws/aws-cryptographic-material-providers-library/pull/1442/commits/4e5d7afd3f90576ce45a315add540a053ee6e6ea)